snuba/replacer.py

```diff
@@ -381,3 +417,85 @@ def flush_batch(self, batch: Sequence[Replacement]) -> None:
             logger.info(
                 "Optimized %s partitions on %s" % (num_dropped, clickhouse_read.host)
             )
+
+    def _message_already_processed(self, metadata: ReplacementMessageMetadata) -> bool:
+        """
+        Figure out whether or not the message was already processed.
+
+        Check if there exists a recorded offset that took too long to execute.
+        If there is, we can conclude whether or not to process the incoming
+        message by comparing the incoming message to the recorded offset.
+        """
+        key = self._build_topic_group_index_key(metadata)
+        self._reset_offset_check(key)
```

Comment:
You actually cannot delete it just yet at this point. 
You would have to wait for a successful commit before deleting. If the consumer restarts and fails again at committing you would have removed the key. 
So you would have to delete only after the commit.
I think that may be more complicated than letting the keys expire, probably not worth trying to delete.

Reply:
This method isn't mean to remove the key every time, it's only for the runtime config fix for when we need to manually roll back the offset in Kafka.

If you take a look at the method you'll see it only resets if necessary:
https://github.com/getsentry/snuba/blob/3737fe7a89f78c3882f351be1377c0f4a278f1e9/snuba/replacer.py#L464-L468